### ISSUE TITLE
Is it still correct to use HttpClientHandler in .Net 6?

### ISSUE THREAD

**Issue by jannavarro** [grpc/grpc-dotnet#1961]
In .Net Core 2.2 and up the SocketsHttpHandler is the preferred HttpHandler to use when configuring channel options HttpHandler. In fact, .Net 6 documentation always show `SocketsHttpHandler` as examples in performance best-practices such as setting `EnableMultipleHttp2Connections` for [connection concurrency](https://learn.microsoft.com/en-us/aspnet/core/grpc/performance?view=aspnetcore-6.0#connection-concurrency).

However, what confuses me is if the fact that `HttpClientHandler` just uses `SocketsHttpHandler` behind the scenes as seen in this [code](https://github.com/dotnet/runtime/blob/f518b2e533ba9c5ed9c1dce3651a77e9a1807b8b/src/libraries/System.Net.Http/src/System/Net/Http/HttpClientHandler.cs#L16).

1. If `HttpClientHandler` just uses `SocketsHttpHandler`, is the code below that sets `a.HttpHandler = new HttpClientHandler` just the same as `a.HttpHandler = new SocketsHttpHandler`?
```
                    services.AddCodeFirstGrpcClient<IService>(
                    config =>
                    {
                        config.Address = //URI;

                        config.ChannelOptionsActions.Add(
                            a =>
                            {
                                a.HttpHandler = new HttpClientHandler
                                {
                                    // Do some setup
                                };
                            });
                    });
```

2. Since `HttpClientHandler` just uses `SocketsHttpHandler` behind the scenes, is there a way to set `EnableMultipleHttp2Connections` when using just `HttpClientHandler` just like in the code above?

**Comment by JamesNK** [comment]
> Since HttpClientHandler just uses SocketsHttpHandler behind the scenes, is there a way to set EnableMultipleHttp2Connections when using just HttpClientHandler just like in the code above?

No, there isn't. That's a reason why to prefer SocketsHttpHandler where possible.
